composeml/label_maker.py

```diff
@@ -232,21 +254,19 @@ def slice(self,
         if num_examples_per_instance == -1:
             num_examples_per_instance = float('inf')
 
-        for key, df in df.groupby(self.target_entity):
-            slices = self._get_slices(df=df, gap=gap, min_data=minimum_data, drop_empty=drop_empty)
+        for key, instance in df.groupby(self.target_entity):
+            slices = self._get_slices(df=instance, gap=gap, min_data=minimum_data, drop_empty=drop_empty)
 
-            for df_slice, df_metadata in slices:
-                df_metadata[self.target_entity] = key
+            for df in slices:
+                df.context.target_entity = self.target_entity
```

Comment:
any way to move this context setting logic into `_get_slices`? I think it'd be better to have that function return the slices all ready to go

Reply:
Yes, will move in `_get_slices`.